### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by krakjoe**
For 7/7.1

### CODE REVIEW

File: tests/tideways_039_prepend.php [hunk 1]
```diff
@@ -1,3 +1,5 @@
 <?php
 
-echo "Hello Auto Prepend!";
+tideways_enable();
+echo "Hello Auto Prepend!\n";
```
Comment (krakjoe) [inline tests/tideways_039_prepend.php]: I just got confused here...

File: tideways.c [hunk 2]
```diff
@@ -3376,7 +3376,7 @@ PHP_FUNCTION(tideways_span_callback)
 		Z_TRY_ADDREF(fci.function_name);
 
 		if (fci.object != NULL) {
-			fci.object->gc.refcount++;
+			GC_REFCOUNT(fci.object)++;
```
Comment (krakjoe) [inline tideways.c]: The internal details of this may change (however unlikely), this is just more forward compatible ...

File: tideways.c [hunk 3]
```diff
@@ -159,8 +159,8 @@ static zend_always_inline void zend_string_release(zend_string *s)
 #define _add_assoc_string_ex(arg, key, key_len, str, copy) add_assoc_string_ex(arg, key, key_len-1, str)
 #define _add_assoc_stringl(arg, key, str, str_len, copy) add_assoc_stringl(arg, key, str, str_len)
 #define _zend_read_property(scope, object, name, name_length, silent, zv) zend_read_property(scope, object, name, name_length, silent, zv)
-#define _call_user_function_ex(function_table, object, function_name, retval_ptr) call_user_function_ex(function_table, object, function_name, retval_ptr, 0, NULL, 1, NULL TSRMLS_CC)
-#define _DECLARE_ZVAL(name) zval name ## _v; zval * name = &name ## _v
+#define tw_call_user_function_ex(function_table, object, function_name, retval_ptr) call_user_function_ex(function_table, object, function_name, retval_ptr, 0, NULL, 1, NULL)
```
Comment (krakjoe) [inline tideways.c]: master (7.1) has _call_user_function_ex symbol ...

File: tideways.c [hunk 4]
```diff
@@ -1412,7 +1412,7 @@ long tw_trace_callback_curl_exec(char *symbol, zend_execute_data *data TSRMLS_DC
 	zval params[1];
 	ZVAL_RES(&params[0], Z_RES_P(argument));
 
-	if (SUCCESS == call_user_function_ex(EG(function_table), NULL, &fname, retval_ptr, 1, params, 1, NULL TSRMLS_CC)) {
+	if (SUCCESS == call_user_function_ex(EG(function_table), NULL, &fname, retval_ptr, 1, params, 1, NULL)) {
```
Comment (krakjoe) [inline tideways.c]: TSRMLS_JUNK isn't necessary in PHP 7 specific code, I didn't have to remove that ... though you may want to adopt omitting it :)

File: tideways.c [hunk 5]
```diff
@@ -3305,11 +3305,11 @@ PHP_FUNCTION(tideways_span_watch)
...
 {
 	tw_watch_callback *twcb = (tw_watch_callback*)Z_PTR_P(zv);
-	if (&twcb->fci.function_name) {
+	if (Z_TYPE(twcb->fci.function_name) != IS_UNDEF) {
 		zval_ptr_dtor(&twcb->fci.function_name);
 	}
-	if (&twcb->fci.object) {
-		zval_ptr_dtor(&twcb->fci.object);
+	if (twcb->fci.object) {
+		zend_object_release(twcb->fci.object);
```
Comment (krakjoe) [inline tideways.c]: zend_fcall_info.object is zend_object* not zval

File: tideways.c [hunk 6]
```diff
@@ -3263,7 +3263,7 @@ void tideways_error_cb(int type, const char *error_filename, const uint error_li
 #else
 static void tideways_throw_exception_hook(zval *exception TSRMLS_DC)
 {
-	zval *exception_ce, *ex;
+	zend_class_entry *exception_ce, *ex;
```
Comment (krakjoe) [inline tideways.c]: *ex appears unused

File: tideways.c (hunk 3 above)
Comment (beberlei) [inline tideways.c]: that reminds me that we should do more automatic testing against master :-)

File: tideways.c (hunk 4 above)
Comment (beberlei) [inline tideways.c]: Right that makes sense

File: tideways.c (hunk 6 above)
Comment (beberlei) [inline tideways.c]: Good catch. I don't remember exception_ce and ex usage here, it is indeed unused. I will cleanup in another commit.

**Pull request merged**